Python/chapter04/p01_route_between_nodes/miguelHx.py

```diff
@@ -0,0 +1,202 @@
+"""Python Version 3.9.2
+4.1 - Route Between Nodes:
+Given a directed graph, design an algorithm to
+find out whether there is a route between two nodes.
+"""
+import unittest
+
+from collections import deque
+from dataclasses import dataclass
+from typing import List, Deque
+
+
+@dataclass
+class Graph:
+    nodes: 'List[Node]'
+
+    def print_graph(self):
+        for node in self.nodes:
+            node.print_children()
+
+    def reset_visited(self):
+        for node in self.nodes:
+            node.visited = False
+
+
+@dataclass
+class Node:
+    id: int
+    children: 'List[Node]'
+    visited: bool = False
+
+    def add_child(self, *nodes: 'Node'):
+        for node in nodes:
+            self.children.append(node)
+
+    def print_children(self):
+        message = f'Adjacency list for node ({self.id}): '
+        for child in self.children:
+            message += '{}, '.format(child.id)
+        print(message)
+
+    def __str__(self):
+        return f'Node ({self.id}), visited: {self.visited}'
+
+
+def dfs_search(root: Node) -> List[int]:
```

Comment:
is this part of the solution? if not, it can be removed.

Reply:
39c084ce4086aee21900712cd5fd9e7574e0ddd0